### PR TITLE
support env assignment and mutable variable assignment with `if` block and `match` guard

### PR DESCRIPTION
**Pull request by WindSoilder**
# Description

Fixes: https://github.com/nushell/nushell/issues/9595

So we can do the following in nushell:
```nushell
mut a = 3
$a = if 4 == 3 { 10 } else {20}
```
or
```nushell
$env.BUILD_EXT = match 3 { 1 => { 'yes!' }, _ => { 'no!' } }
```

# User-Facing Changes

# Tests + Formatting

# After Submitting

### CODE REVIEW

**Comment by sholderbach** [comment]
Although I don't like the need to add another special case, this should be something that works. Thanks for figuring it out and covering it with the happy path tests necessary!

**Comment by sholderbach** [comment]
Maybe we can reconcile those changes with the work @1Kinoti is doing in #9651 ?

**Comment by 1Kinoti** [comment]
mmmh, #9651 only works with `let`, `mut` and `const`.

i don't know how we would make it work with variable mutation 🤔. but i'll look into it

File: crates/nu-command/tests/commands/mut_.rs [hunk 1]
```diff
@@ -177,3 +177,17 @@ fn mut_records_update_properly() {
     let actual = nu!(pipeline("mut a = {}; $a.b.c = 100; $a.b.c"));
     assert_eq!(actual.out, "100");
 }
+
+#[test]
+fn mut_value_with_if() {
+    let actual = nu!(pipeline("mut a = 3; $a = if 3 == 3 { 10 }; echo $a"));
```
Comment (amtoine) [inline crates/nu-command/tests/commands/mut_.rs:183]: as the commands for these 4 tests are on single lines, i'm not sure we need to wrap them inside `pipeline` :thinking:

File: tests/shell/environment/env.rs [hunk 2]
```diff
@@ -86,6 +86,22 @@ fn env_assignment() {
     assert_eq!(actual.out, "barbaz");
 }
 
+#[test]
+fn env_assignment_with_if() {
+    let actual = nu!(cwd: ".", r#"
```
Comment (amtoine) [inline tests/shell/environment/env.rs:91]: i do not think we need these two `cwd` here :yum:

**Review by amtoine — CHANGES_REQUESTED** [review]
i've tested this PR and it appears to fix the issue, very nice job @WindSoilder 

i've got a few remarks on the tests, otherwise, i'm ready to approve the changes :yum:

File: crates/nu-command/tests/commands/mut_.rs (hunk 1 above)
Comment (WindSoilder) [reply crates/nu-command/tests/commands/mut_.rs]: Thanks for catching this!

File: tests/shell/environment/env.rs (hunk 2 above)
Comment (WindSoilder) [reply tests/shell/environment/env.rs]: Yeah I think so, thanks for catching this!

**Comment by sholderbach** [comment]
Thanks @WindSoilder I think we can land this now. @1Kinoti 's work looks unconnected for now, we can refactor this later.

**Pull request merged**